torch/nn/functional.py

```diff
@@ -2777,6 +2777,55 @@ def pairwise_distance(x1, x2, p=2., eps=1e-6, keepdim=False):
 """)
 
 
+one_hot = _add_docstr(torch._C._nn.one_hot, r"""
+one_hot(tensor, num_classes=0) -> LongTensor
+
+Takes LongTensor with index values of shape ``(*)`` and returns a tensor
+of shape ``(*, num_classes)`` that have zeros everywhere except where the
+index of last dimension matches the corresponding value of the input tensor,
+in which case it will be 1.
+
+See also `One-hot on Wikipedia`_ .
+
+.. _One-hot on Wikipedia:
+    https://en.wikipedia.org/wiki/One-hot
+
+Arguments:
+    tensor (LongTensor): class values of any shape.
+    num_classes (int):  Total number of classes. If set to -1, the actually
+        number of classes will be inferred as the largest class value in the
```

Comment:
"the actual number of classes is inferred to be one greater than the largest class value" - the "one greater than" is important